sktime/forecasting/arch/_uarch.py

```diff
@@ -0,0 +1,362 @@
+        Parameters required to forecast. Must be identical in shape to the
+        parameters computed by fitting the model.
+    start : {int, datetime, Timestamp, str}, optional
+        An integer, datetime or str indicating the first observation to produce the
+        forecast for. Datetimes can only be used with pandas inputs that have a
+        datetime index. Strings must be convertible to a date time, such as in
+        '1945-01-01'.
+    align : str, optional
+        Either 'origin' or 'target'. When set of 'origin', the t-th row of forecast
+        contains the forecasts for t+1, t+2, ..., t+h. When set to 'target', the
+        t-th row contains the 1-step ahead forecast from time t-1, the 2 step from
+        time t-2, ..., and the h-step from time t-h. 'target' simplified computing
+        forecast errors since the realization and h-step forecast are aligned.
+    method : {'analytic', 'simulation', 'bootstrap'}
+        Method to use when producing the forecast. The default is analytic.
+        The method only affects the variance forecast generation. Not all volatility
+        models support all methods.
+        In particular, volatility models that do not evolve in squares such as
+        EGARCH or TARCH
+        do not support the 'analytic' method for horizons > 1.
+    simulations : int
+        Number of simulations to run when computing the forecast using either
+        simulation or bootstrap.
+    rng : callable, optional
+        Custom random number generator to use in simulation-based forecasts.
+        Must produce random samples using the syntax rng(size) where size the
+        2-element tuple (simulations, horizon).
+    random_state : RandomState, optional
+        NumPy RandomState instance to use when method is 'bootstrap'
+    reindex : bool, optional
+        Whether to reindex the forecasts to have the same dimension as the series
+        being forecast. Prior to 4.18 this was the default. As of 4.19 this is now
+        optional. If not provided, a warning is raised about the future change in
+        the default which will occur after September 2021.
+
+    See Also
+    --------
+    StatsForecastARCH
+    StatsForecastGARCH
+
+    References
+    ----------
+    .. [1] Jason Brownlee. How to Model Volatility with ARCH and GARCH for Time Series
+       Forecasting.
+       https://machinelearningmastery.com/develop-arch-and-garch-models-for-time-series-
+       forecasting-in-python/
+    .. [2] GitHub repository of arch package (soft dependency).
+       https://github.com/bashtage/arch
+    .. [3] Documentation of arch package (soft dependency). Forecasting Volatility with
+       ARCH and it's variants.
+       https://arch.readthedocs.io/en/latest/univariate/introduction.html
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.garch import ARCH
+    >>> y = load_airline()
+    >>> forecaster = ARCH()
```

Comment:
When doctests are run, the dependencies are not installed, at least not for all jobs. You have to tell doctest to skip this example to avoid CI failures.

Reply:
@Vasudeva-bit tge current CI failure is coming from this line. Use `# doctest: +SKIP` in this like and the next two dependent ones.

Ref. https://docs.pytest.org/en/stable/how-to/doctest.html#skipping-tests